src/components/expired-subscription-modal/index.jsx

```diff
@@ -31,15 +18,21 @@ const ExpiredSubscriptionModal = () => {
       footerNode={(
         <ActionRow>
           <Button
-            onClick={onClickHandler}
+            onClick={() => window.open(customerAgreement?.urlForButtonInModal, '_blank')}
           >
             {customerAgreement?.buttonLabelInModal}
           </Button>
         </ActionRow>
       )}
     >
       {/* eslint-disable-next-line react/no-danger */}
-      <div dangerouslySetInnerHTML={{ __html: customerAgreement?.expiredSubscriptionModalMessaging }} />
+      <div dangerouslySetInnerHTML={{
+        __html: DOMPurify.sanitize(
+          customerAgreement?.expiredSubscriptionModalMessaging,
```

Comment:
nit: `?.` can be removed due to the early return above (removing confusion around whether `customerAgreement` might be undefined/null in this code path.